src/main/scala/com/rbmhtechnology/calliope/scaladsl/TransactionalEventProducer.scala

```diff
@@ -0,0 +1,257 @@
+
+    private[calliope] def producerSettings[A](): ProducerSettings[String, SequencedMessage[A]] =
+      kafkaProducerSettings.asInstanceOf[ProducerSettings[String, SequencedMessage[A]]]
+
+    private[calliope] def producer[A](): KafkaProducer[String, SequencedMessage[A]] =
+      kafkaProducer.asInstanceOf[KafkaProducer[String, SequencedMessage[A]]]
+
+    def withReadBufferSize(readBufferSize: Int): Settings =
+      copy(readBufferSize = readBufferSize)
+
+    def withReadInterval(readInterval: FiniteDuration): Settings =
+      copy(readInterval = readInterval)
+
+    def withDeleteInterval(deleteInterval: FiniteDuration): Settings =
+      copy(deleteInterval = deleteInterval)
+
+    def withTransactionTimeout(transactionTimeout: FiniteDuration): Settings =
+      copy(transactionTimeout = transactionTimeout)
+
+    def withBootstrapServers(bootstrapServers: String): Settings =
+      copy(kafkaProducerSettings = kafkaProducerSettings.withBootstrapServers(bootstrapServers))
+
+    def withProducerCloseTimeout(closeTimeout: FiniteDuration): Settings =
+      copy(kafkaProducerSettings = kafkaProducerSettings.withCloseTimeout(closeTimeout))
+
+    def withProducerDispatcher(dispatcher: String): Settings =
+      copy(kafkaProducerSettings = kafkaProducerSettings.withDispatcher(dispatcher))
+
+    def withProducerParallelism(parallelism: Int): Settings =
+      copy(kafkaProducerSettings = kafkaProducerSettings.withParallelism(parallelism))
+
+    def withProducerProperty(key: String, value: String): Settings =
+      copy(kafkaProducerSettings = kafkaProducerSettings.withProperty(key, value))
+
+    private def copy(readBufferSize: Int = readBufferSize,
+                     readInterval: FiniteDuration = readInterval,
+                     deleteInterval: FiniteDuration = deleteInterval,
+                     transactionTimeout: FiniteDuration = transactionTimeout,
+                     kafkaProducerSettings: ProducerSettings[String, SequencedMessage[AnyRef]] = kafkaProducerSettings): Settings =
+      new Settings(readBufferSize, readInterval, deleteInterval, transactionTimeout, kafkaProducerSettings)
+  }
+}
+
+class TransactionalEventProducer[A] private(sourceId: String, topic: String, eventStore: EventStore[A], onFailure: FailureHandler, settings: Settings)
+                                           (implicit val system: ActorSystem) extends IoDispatcher {
+
+  import ProducerGraphRunner._
+
+  private var running = false
+
+  private val graph = ProducerGraph(
+    EventStoreReader.withTimestampGapDetection(sourceId, eventStore, settings.transactionTimeout),
+    new EventStoreDeleter(eventStore),
+    settings)
+
+  private lazy val graphRunner = system.actorOf(Props(new GraphRunnerSupervisor(graph, onFailure)))
+
+  def run(): EventWriter[A] = {
+    running = true
+    val runner = graphRunner
+    new EventStoreWriter[A](topic, eventStore, _ => runner ! NotifyCommit)
```

Comment:
if used directly, the graphRunner will only be started on the first commit (as it is a lazy val). By assigning it here the initialization is actually performed in the `run` method.